CombinatorialGames/Nimber/Simplicity.lean

```diff
@@ -0,0 +1,289 @@
+  coefficients `< x`.
+
+## Todo
+
+We are currently at 1/4.
+-/
+
+open Ordinal Polynomial Set
+
+/-! ### Mathlib lemmas -/
+
+theorem le_of_forall_ne {α : Type*} [LinearOrder α] {x y : α} (h : ∀ z < x, z ≠ y) : x ≤ y := by
+  contrapose! h
+  use y
+
+theorem lt_add_iff_lt_or_exists_lt {α : Type*} [Add α] [LinearOrder α] [CanonicallyOrderedAdd α]
+    [AddLeftReflectLT α] [IsLeftCancelAdd α] {a b c : α} :
+    a < b + c ↔ a < b ∨ ∃ d < c, a = b + d := by
+  obtain h | h := lt_or_ge a b
+  · have : a < b + c := h.trans_le (le_self_add ..)
+    tauto
+  · obtain ⟨a, rfl⟩ := exists_add_of_le h
+    simp
+
+theorem forall_lt_add {α : Type*} [Add α] [LinearOrder α] [CanonicallyOrderedAdd α]
+    [AddLeftReflectLT α] [IsLeftCancelAdd α] {b c : α} {P : α → Prop} :
+    (∀ a < b + c, P a) ↔ (∀ a < b, P a) ∧ (∀ a < c, P (b + a)) := by
+  simp_rw [lt_add_iff_lt_or_exists_lt]
+  aesop
+
+theorem exists_lt_add {α : Type*} [Add α] [LinearOrder α] [CanonicallyOrderedAdd α]
+    [AddLeftReflectLT α] [IsLeftCancelAdd α] {b c : α} {P : α → Prop} :
+    (∃ a < b + c, P a) ↔ (∃ a < b, P a) ∨ (∃ a < c, P (b + a)) := by
+  simp_rw [lt_add_iff_lt_or_exists_lt]
+  aesop
+
+theorem le_add_iff_lt_or_exists_le {α : Type*} [Add α] [LinearOrder α] [CanonicallyOrderedAdd α]
+    [AddLeftMono α] [IsLeftCancelAdd α] {a b c : α} :
+    a ≤ b + c ↔ a < b ∨ ∃ d ≤ c, a = b + d := by
+  obtain h | h := lt_or_ge a b
+  · have : a ≤ b + c := h.le.trans (le_self_add ..)
+    tauto
+  · obtain ⟨a, rfl⟩ := exists_add_of_le h
+    simp
+
+theorem forall_le_add {α : Type*} [Add α] [LinearOrder α] [CanonicallyOrderedAdd α]
+    [AddLeftMono α] [IsLeftCancelAdd α] {b c : α} {P : α → Prop} :
+    (∀ a ≤ b + c, P a) ↔ (∀ a < b, P a) ∧ (∀ a ≤ c, P (b + a)) := by
+  simp_rw [le_add_iff_lt_or_exists_le]
+  aesop
+
+theorem exists_le_add {α : Type*} [Add α] [LinearOrder α] [CanonicallyOrderedAdd α]
+    [AddLeftMono α] [IsLeftCancelAdd α] {b c : α} {P : α → Prop} :
+    (∃ a ≤ b + c, P a) ↔ (∃ a < b, P a) ∨ (∃ a ≤ c, P (b + a)) := by
+  simp_rw [le_add_iff_lt_or_exists_le]
+  aesop
+
+namespace Ordinal
+
+theorem div_two_opow_log {o : Ordinal} (ho : o ≠ 0) : o / 2 ^ log 2 o = 1 := by
+  apply le_antisymm
+  · simpa [← succ_one] using div_opow_log_lt o one_lt_two
+  · simpa [← succ_zero] using div_opow_log_pos 2 ho
+
+theorem two_opow_log_add {o : Ordinal} (ho : o ≠ 0) : 2 ^ log 2 o + o % 2 ^ log 2 o = o := by
+  convert div_add_mod .. using 2
+  rw [div_two_opow_log ho, mul_one]
+
+protected theorem mul_two (o : Ordinal) : o * 2 = o + o := by
+  rw [← one_add_one_eq_two, mul_add, mul_one]
+
+theorem lt_mul_iff {a b c : Ordinal} : a < b * c ↔ ∃ q < c, ∃ r < b, a = b * q + r := by
+  obtain rfl | hb₀ := eq_or_ne b 0; simp
+  refine ⟨fun h ↦ ⟨_, (Ordinal.div_lt hb₀).2 h, _, mod_lt a hb₀, (div_add_mod ..).symm⟩, ?_⟩
+  rintro ⟨q, hq, r, hr, rfl⟩
+  apply (add_left_strictMono hr).trans_le
+  simp_rw [← mul_succ]
+  apply mul_le_mul_left'
+  simpa
```

Comment:
`simpa` seems to be being used in your recent PRs as a kind of embedded `exact?` - is this preferred in mathlib?

```suggestion
  exact Order.succ_le_iff.mpr hq
```

Reply:
`exact` is certainly preferred in situations like this, I think I just forgot to clean that one up.